src/themes/teams/components/Icon/iconVariables.ts

```diff
@@ -1,31 +1,39 @@
+import { ColorValues } from '../../../types'
+import { mapColorsToScheme } from '../../../../lib'
 import { pxToRem } from '../../utils'
 
-export interface IconVariables {
-  [key: string]: string | number | boolean | undefined
-
-  redColor?: string
-  brandColor?: string
+export type IconSizeModifier = 'x' | 'xx'
 
-  outline?: boolean
+export interface IconVariables {
+  colors: ColorValues<string>
   color?: string
   backgroundColor?: string
   borderColor?: string
-  horizontalSpace: string
-  margin: string
+  brandColor?: string
   secondaryColor: string
+  redColor?: string
   disabledColor: string
+
+  outline?: boolean
+  horizontalSpace: string
+  margin: string
+  sizeModifier: IconSizeModifier
 }
 
+const colorVariant = 500
+
 export default (siteVars): IconVariables => ({
-  outline: undefined,
+  colors: mapColorsToScheme(siteVars, colorVariant),
   color: undefined,
   backgroundColor: undefined,
   borderColor: undefined,
-  horizontalSpace: pxToRem(10),
-  margin: `0 ${pxToRem(8)} 0 0`,
+  brandColor: siteVars.brandColor,
   secondaryColor: siteVars.white,
+  redColor: siteVars.red,
   disabledColor: siteVars.gray06,
 
-  redColor: siteVars.red,
-  brandColor: siteVars.brandColor,
+  outline: undefined,
+  horizontalSpace: pxToRem(10),
+  margin: `0 ${pxToRem(8)} 0 0`,
+  sizeModifier: undefined,
```

Comment:
Is there are real need to define `undefined` explicitly?

Reply:
not really, made it optional and removed from values